lib/configDefaults.js

```diff
@@ -1,7 +1,30 @@
 const os = require('os')
 const path = require('path')
+const fs = require('fs')
 
-module.exports = {
+const reporter = require('./reporter')
+
+const userConfig = () => {
+    const configDir = path.join(os.homedir(), '.config', 'd2')
+    const configPattern = /^(d2\.)*(config)\.(js$|json$)/
+
+    try {
+        const files = fs.readdirSync(configDir)
+
+        // grab one of the config files if there are more than one
+        const configFile = files.filter(f => configPattern.test(f))[0]
+
+        const userConfig = require(path.join(configDir, configFile))
+        reporter.debug(`User configuration loaded from: ${configFile}`, userConfig)
+
+        return userConfig
+    } catch (e) {
+        reporter.debug(`Failed to load user config from: ${configDir}`, e)
+        return {}
```

Comment:
If the `try` block fails for any reason, we default to an empty user config.

Reply:
I think it's fine, just a bit opaque if the file doesn't exist and you run with `--verbose`, you'll see `Failed to load` messages.